mt-kahypar/partition/refinement/fm/localized_kway_fm_core.cpp

```diff
@@ -247,28 +249,54 @@ namespace mt_kahypar {
           PartitionedHypergraph& phg,
           const size_t best_index_locally_observed,
           const Gain best_improvement_locally_observed,
-          bool apply_all_moves) {
+          const bool apply_delta_improvement) {
 
     Gain improvement_from_attributed_gains = 0;
     Gain attributed_gain = 0;
+    bool is_last_move = false;
 
     auto delta_gain_func = [&](const HyperedgeID he,
-                               const HyperedgeWeight edge_weight,
-                               const HypernodeID edge_size,
-                               const HypernodeID pin_count_in_from_part_after,
-                               const HypernodeID pin_count_in_to_part_after) {
+                                    const HyperedgeWeight edge_weight,
+                                    const HypernodeID edge_size,
+                                    const HypernodeID pin_count_in_from_part_after,
+                                    const HypernodeID pin_count_in_to_part_after) {
       attributed_gain += km1Delta(he, edge_weight, edge_size,
                                   pin_count_in_from_part_after, pin_count_in_to_part_after);
+
+      // Gains of the pins of a hyperedge can only change in the following situations.
+      if ( is_last_move &&
+           ( pin_count_in_from_part_after == 0 || pin_count_in_from_part_after == 1 ||
+             pin_count_in_to_part_after == 1 || pin_count_in_to_part_after == 2 ) ) {
+        edgesWithGainChanges.push_back(he);
```

Comment:
add comment that this vector is not cleared before acquireNeighbors is called. In case this ever changes